### PR TITLE
FFI: add function rnp_key_valid_till64()

### PR DESCRIPTION
**Pull request by ni4**
Currently function `rnp_key_valid_till()` returns uint32_t, which is good enough only till the year 2038.
While it is very unlikely that some implementation would produce keys which expire after the year 2038 (and actually till the year 2038 OpenPGP standard should be updated to correctly handle creation time), we still should have some interface for the future.

Fixes issue #1480.

### CODE REVIEW

File: src/lib/rnp.cpp [hunk 1]
```diff
@@ -6421,6 +6421,28 @@ FFI_GUARD
...
+    }
+    uint64_t     res = 0;
+    rnp_result_t ret = rnp_key_valid_till64(handle, &res);
+    if (ret) {
+        return ret;
+    }
+    if (res == UINT64_MAX) {
+        *result = UINT32_MAX;
+    } else if (res > UINT32_MAX) {
+        *result = UINT32_MAX - 1;
```
Comment (nwalfield) [inline src/lib/rnp.cpp:6436]: Why the `- 1`?

Comment (ni4) [reply src/lib/rnp.cpp:6436]: Because UINT32_MAX is reserved for 'never expires'.

Comment (nwalfield) [reply src/lib/rnp.cpp:6436]: Then you should probably do: `if (res >= UINT32_MAX)` to catch the case that `res` happens to equal `UINT32_MAX`.

Comment (ni4) [reply src/lib/rnp.cpp:6436]: Agree, missed that. Thanks.

**Review by antonsviridenko — APPROVED** [review]

**Review by ronaldtse — APPROVED** [review]

**Comment by ronaldtse** [comment]
Thank you all!

**Pull request merged**